xds/internal/resolver/xds_resolver.go

```diff
@@ -119,6 +119,7 @@ func (b *xdsResolverBuilder) Build(target resolver.Target, cc resolver.ClientCon
 		endpoint = target.URL.Opaque
 	}
 	endpoint = strings.TrimPrefix(endpoint, "/")
+	r.dataplaneAuthority = endpoint
```

Comment:
This isn't the "last path component". If you have xds:///foo/bar then the string here is "foo/bar".

Reply:
Thanks for bringing this up. Just by intuition I assumed that the dataplane authority (or the virtual host domain name to lookup for) should be the endpoint from the dial target. 

I brought this up with the gRPC xDS group and Mark has a [patchset](https://github.com/grpc/proposal/pull/420) to the A47 gRFC to clarify this. 

Given that I think this is the expected behavior.

Reply:
I agree they should be the same, but the docs you copied said it was different than you were using. (And the docs did say they were the same.)

Reply:
:) yes, comment needs updating.. I'll fix the comment once there is an LGTM on the doc update

Reply:
I don't see "any remaining '/' characters will be percent-encoded" implemented here.

Reply:
I did some investigation and it seems like we are percentage encoding everything in the URI path except for `/`s _intentionally_ (see [here](https://github.com/grpc/grpc-go/blob/5ccf176a08abbb41c17c5de5930aa5bb18f2563b/xds/internal/xdsclient/bootstrap/template.go#L40)). I have opened https://github.com/grpc/grpc-go/issues/7002 to track the investigation.

I've updated the PR to only change the virtualhost lookup logic and will track the % encoding question as a followup.

Reply:
I was able to track #7002 and close it. Basically xDS URI targets use the default way to determine channel authority which is to %encoding the path component. 

And yes I verified that we %encoding `/`s after https://github.com/grpc/grpc-go/commit/fc0aa4689cd434c21cd4cbbbe9ef5af842ccb326 was merged

Reply:
Currently, there is an open question about whether or not to use the %-encoded authority to match the virtual host domain.

per @markdroth - 
C-core currently uses the %-encoded authority to match the virtual host domain.

@ejona86 , @dfawley -- Do you have a strong preference or shall we be guided by the existing implementation?

Reply:
Per @ejona86 --
We will send the percent-encoded form on the wire, so it seems appropriate to match against the same here. That's what the server would match against.

In general, the fewer different representations of authority, the fewer chances for subtle bugs.